incubator/mysqlha/templates/statefulset.yaml

```diff
@@ -215,6 +215,35 @@ spec:
           requests:
             cpu: 100m
             memory: 100Mi
+      {{- if .Values.metrics.enabled }}
```

Comment:
since the metrics endpoint will report the global status of the cluster, I am not sure if we should add a exporter for each pod as opposed to having a single exporter that talks to the master node to gather the metrics.

Reply:
There are some slave-specific metrics that you might want to analyse so only running on master feels counterproductive.

Reply:
There are some slave-specific metrics that you might want to analyse so only running on master feels counterproductive.

Reply:
I am not very knowledgeable of the metrics the mysql exporter exposes, but looking at the documentation at https://github.com/prometheus/mysqld_exporter it appears a config named `collect.slave_status` (Collect from SHOW SLAVE STATUS) is enabled by default. That's the reason I was wondering if it really made sense to have as many mysqld_exporters as there are mysqld pods. 

Additionally in svc.yaml the metrics port is exposed only for the `write` endpoint (service with the name `{{ template "fullname" . }}`) which means that the exporters associated with the readonly pods will not be reached anyway.

@jpoon in svc.yaml i see two services exist `{{ template "fullname" . }}` and `{{ template "fullname" . }}-readonly`. i was wondering how the pods are filtered here.

Reply:
As slaves are used for read request, you want to gather metrics from them separately. Metrics exposed are things you'd use if you run into performance problems, for instance. For that, they need to be available per instance.

Reply:
To add to this, `collect.slave_status` just adds the metrics it can collect from SHOW SLAVE STATUS, but that's mostly just how much lag a slave has with the master.